src/openms/source/CHEMISTRY/SimpleTSGXLMS.cpp

```diff
@@ -321,7 +326,13 @@ namespace OpenMS
       addPrecursorPeaks_(spectrum, precursor_mass, maxcharge);
     }
 
-    std::sort(spectrum.begin(), spectrum.end(), SimpleTSGXLMS::SimplePeakComparator());
+#ifdef OPENMS_USE_PDQSORT
+    std::reverse(spectrum.begin(), spectrum.end());
+    boost::sort::pdqsort(spectrum.begin(), spectrum.end(), [](const SimplePeak& a, const SimplePeak& b) {return a.mz < b.mz;});
```

Comment:
would spectrum.rbegin() work instead of using std::reverse?

Reply:
then I would sort them in reverse order, descending, and would have to reverse them again afterwards. although it might actually work if I reverse the comparator...  I will try something

Reply:
yep comp needs to be reversed

Reply:
ok, it works, but for some reason is slower. benchmark went from 34s to 43s by using reverse iterators and a reversed comparator. I think using rbegin and rend with a reversed comparator essentially does the same operation, just starting from a different end. So the pre-sortedness of the spectra still only helps, if you reverse it first.